scm/driver/github/milestone.go

```diff
@@ -0,0 +1,111 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/drone/go-scm/scm"
+)
+
+type milestoneService struct {
+	client *wrapper
+}
+
+type milestone struct {
+	ID           int       `json:"id"`
+	Number       int       `json:"number"`
+	Title        string    `json:"title"`
+	Description  string    `json:"description"`
+	State        string    `json:"state"`
+	DueOn        time.Time `json:"due_on"`
+	URL          string    `json:"url"`
+	HTMLURL      string    `json:"html_url"`
+	LabelsURL    string    `json:"labels_url"`
+	Creator      user      `json:"creator"`
+	OpenIssues   int       `json:"open_issues"`
+	ClosedIssues int       `json:"closed_issues"`
+	NodeID       string    `json:"node_id"`
+	CreatedAt    time.Time `json:"created_at"`
+	UpdatedAt    time.Time `json:"updated_at"`
+	ClosedAt     time.Time `json:"closed_at"`
+}
+
+type milestoneInput struct {
+	Title       string    `json:"title"`
+	State       string    `json:"state"`
+	Description string    `json:"description"`
+	DueOn       time.Time `json:"due_on"`
+}
+
+func (s *milestoneService) Find(ctx context.Context, repo string, id int) (*scm.Milestone, *scm.Response, error) {
+	path := fmt.Sprintf("repos/%s/milestones/%d", repo, id)
+	out := new(milestone)
+	res, err := s.client.do(ctx, "GET", path, nil, out)
+	return convertMilestone(out), res, err
+}
+
+func (s *milestoneService) List(ctx context.Context, repo string, opts scm.MilestoneListOptions) ([]*scm.Milestone, *scm.Response, error) {
+	path := fmt.Sprintf("repos/%s/milestones?%s", repo, encodeMilestoneListOptions(opts))
+	out := []*milestone{}
+	res, err := s.client.do(ctx, "GET", path, nil, &out)
+	return convertMilestoneList(out), res, err
+}
+
+func (s *milestoneService) Create(ctx context.Context, repo string, input *scm.MilestoneInput) (*scm.Milestone, *scm.Response, error) {
+	path := fmt.Sprintf("repos/%s/milestones", repo)
+	in := &milestoneInput{
+		Title:       input.Title,
+		State:       input.State,
+		Description: input.Description,
+		DueOn:       *input.DueDate,
```

Comment:
we try to avoid using pointers in the codebase.  Instead we have a `null` package where you can use `null.Time` or `null.String`.  You can search some of the other packages to see how we use it.  Also happy to jump on a call to review.

https://github.com/drone/go-scm/tree/master/scm/driver/internal/null

Reply:
done.